app/models/ability.rb

```diff
@@ -67,7 +67,11 @@ def admin_perms
       obj.team_id == @context_team.id || obj.feed.team_id == @context_team.id
     end
     can [:create, :update, :read, :destroy], ApiKey, :team_id => @context_team.id
-    can :destroy, [Dynamic, DynamicAnnotation::Field], annotation_type: 'metadata' # FIXME: Restrict by team
+    # can :destroy, [Dynamic, DynamicAnnotation::Field], annotation_type: 'metadata' # FIXME: Restrict by team
+    can :destroy, [Dynamic, DynamicAnnotation::Field] do |obj|
+      source = Source.find(obj.id)
+      source.team_id == @context_team.id
```

Comment:
@vasconsaurus , here you're using the `Dynamic` or `DynamicAnnotation::Field` ID to load a `Source`, assuming that the IDs are the same, but they're not (actually even in a 1:1 relationship, this is fragile). I noticed that looks like both `Dynamic` and `DynamicAnnotation::Field` have a `team` method, so I think you can simply do:

```ruby
can :destroy, [Dynamic, DynamicAnnotation::Field] do |obj|
  obj.team.present? && obj.team == @context_team
end
```

Reply:
Updated!